compiler-rt/test/rtsan/blocking_call.cpp

```diff
@@ -7,18 +7,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-// TODO: Remove when [[blocking]] is implemented.
-extern "C" void __rtsan_notify_blocking_call(const char *function_name);
-
-void custom_blocking_function() {
-  // TODO: When [[blocking]] is implemented, don't call this directly.
-  __rtsan_notify_blocking_call(__func__);
+void custom_blocking_function() [[clang::blocking]] {
+  printf("In blocking function\n");
 }
 
-void safe_call() {
-  // TODO: When [[blocking]] is implemented, don't call this directly.
-  __rtsan_notify_blocking_call(__func__);
-}
+void safe_call() [[clang::blocking]] { printf("In safe call\n"); }
```

Comment:
I'm a bit confused as to why we have a function called `safe_call` that's attributed `[[clang::blocking]]`. The test assertions also check that `safe_call` is not printed. Was it originally a mistake to call `__rtsan_notify_blocking_call` in the test when it was first written, maybe?

Reply:
This call is to test that [[blocking]] called from an un-attributed setting (main) does not result in any error. Happy to take suggestions on how to make this more clear!

Reply:
I just took a second crack at it with different assertions etc, let me know what you think

Reply:
Oops - I missed the call to `safe_call` in `main` - my mistake. Anyway, if we wanted to make an improvement, I think we could organise `main` into:

```cpp
int main() {
    nonrealtime_function();
    realtime_function();
    return 0;
}
```

where the wrappers `nonrealtime_function` and `realtime_function` are simply:

```cpp
void realtime_function() [[clang::nonblocking]] { custom_blocking_function(); }
void nonrealtime_function() { custom_blocking_function(); }
```

and notice we have eliminated `safe_call`.